crates/vision/src/image_segmenter.rs

```diff
@@ -315,94 +314,62 @@ fn pixel_to_edge_detection_value(
-                fourth_pixel.cr,
-                fifth_pixel.cr,
-            ]);
-
-            YCbCr444::new(y, cb, cr)
-        }
-        4..=5 => {
-            let length = segment.length();
-            let first_position = segment.start + (length / 4);
-            let second_position = segment.start + ((length * 2) / 4);
-            let third_position = segment.start + ((length * 3) / 4);
-
-            let first_pixel = image.at(x, first_position as u32);
-            let second_pixel = image.at(x, second_position as u32);
-            let third_pixel = image.at(x, third_position as u32);
-
-            let y = median_of_three([first_pixel.y, second_pixel.y, third_pixel.y]);
-            let cb = median_of_three([first_pixel.cb, second_pixel.cb, third_pixel.cb]);
-            let cr = median_of_three([first_pixel.cr, second_pixel.cr, third_pixel.cr]);
-
-            YCbCr444::new(y, cb, cr)
+fn set_color_in_vertical_segment(mut segment: Segment, image: &YCbCr422Image, x: u32) -> Segment {
+    let length = segment.length();
+    let stride = match length {
+        20.. => 4,   // results in 5.. or more sample pixels
+        7..=19 => 2, // results in 4..=10 or more sample pixels
+        1..=6 => 1,  // results in 1..=6 or more sample pixels
+        0 => {
+            segment.color = image.at(x, segment.start as u32);
+            return segment;
         }
-        0..=3 => {
-            let position = segment.start + segment.length() / 2;
-            image.at(x, position as u32)
-        }
-    };
-    segment.color = if segment.length() >= 4 {
-        let spacing = segment.length() / 4;
-        let first_position = segment.start + spacing;
-        let second_position = segment.start + 2 * spacing;
-        let third_position = segment.start + 3 * spacing;
-
-        let first_pixel = image.at(x, first_position as u32);
-        let second_pixel = image.at(x, second_position as u32);
-        let third_pixel = image.at(x, third_position as u32);
-
-        let y = median_of_three([first_pixel.y, second_pixel.y, third_pixel.y]);
-        let cb = median_of_three([first_pixel.cb, second_pixel.cb, third_pixel.cb]);
-        let cr = median_of_three([first_pixel.cr, second_pixel.cr, third_pixel.cr]);
-        YCbCr444::new(y, cb, cr)
-    } else {
-        let position = segment.start + segment.length() / 2;
-        image.at(x, position as u32)
     };
+    let mut pixels = sample_image_pixels_vertically(
+        image,
+        x,
+        (segment.start as u32)..(segment.end as u32),
+        stride,
+    );
+    let length = pixels.len();
+    let (_, median_pixel, _) = pixels.select_nth_unstable_by_key(length / 2, |pixel| pixel.y);
+    segment.color = *median_pixel;
+    segment
+}
+
+fn set_field_color_in_vertical_segment(mut segment: Segment, field_color: &FieldColor) -> Segment {
     segment.field_color = field_color.get_intensity(segment.color);
     segment
 }
 
+fn sample_image_pixels_vertically(
```

Comment:
What do you think about making this a function of the `Image`? To hide a lot of implementation detail, and not making the `coordinates_to_buffer_index` function `pub`.
Maybe consider replacing "vertically" with something like column?

Reply:
- `YCbCr422Image::buffer()` is now private
- `YCbCr422Image::coordinates_to_buffer_index()` is now private
- The Image Color Select panel uses a new public method `YCbCr422Image::iter_pixels()` which returns the 444 pixels
- Iteration in Image Segmenter is now done in coordinate-domain instead of buffer-index-domain